### DISCUSSION TITLE
Extract video resolution

### DISCUSSION THREAD

**Discussion by kisasexypantera94** [pion/webrtc#2285] [General] [answered]
Hi, I have a pion webrtc signaling server which relays videos to transcoder server for rescaling into smaller resolutions - what is the natural way to extract video resolution coming from client using pion? I see several ways but I lack expertise to decide which is better for my case:
1. Depay and decode first RTP packet and peek at metadata
2. Pack video resolution info into rtp packet
3. Peek at sdp?
4. Send as a message to signaler

**Comment by Sean-Der** [comment, marked as answer]
Hey @kisasexypantera94 

One thing to note is the video resolution will dynamically change during the call (if using libwebrtc or other clients with BWE).

1.) This could work! This will be codec specific, but if doing H264 you can get this from the SPS/PPS.

2.) Some RTP Payloaders (AV1) include this in the OBU. Making it even easier then above.

3.) No information in the SDP will be helpful.

4.) You could get the information via `getStats` on the clients and send it via your signaling.

**Reply by kisasexypantera94** [reply]
Thanks for the quick response (and for pion too). For the sake of history, I sticked to a hybrid of depay/decode and getStats options - the first is being executed on the transcoder itself for quick configuration of outgoing streams in lower resolutions and the second is used for more sophisticated logic on signaler side
